api/envoy/api/v2/route/route.proto

```diff
@@ -361,7 +361,25 @@ message RouteAction {
   //
   // .. note::
   //
-  //   As an example, given following config,
+  //   Take a careful attention to the use of trailing slash in match prefix value. A common
+  //   use-case is to have a rule to rewrite e.g. :code:`scheme://host/prefix` and
+  //   :code:`scheme://host/prefix/ok/1/2/3` to the root path of the host (:code:`scheme://host/`)
+  //   and :code:`scheme://host/ok/1/2/3` respectively.
+  //
+  //   The following config example exhibits that behavior.
+  //
+  //   .. code-block:: yaml
+  //
+  //     - match:
+  //         prefix: "/prefix/"
+  //       route:
+  //         prefix_rewrite: "/"
+  //     - match:
+  //         prefix: "/prefix"
+  //       route:
+  //         prefix_rewrite: "/"
+  //
```

Comment:
I think the examples below will be clearer if the two prefix-substitutions above use different match strings.

Reply:
Given:

Reply:
Given:

Reply:
Could you clarify?

Reply:
The point of this is to tell that to achieve such behavior (rewriting replace a segment and attach the rest to the root path) is by declaring two match rules (with "/" and without "/", in this case "/prefix" and "/prefix/"). The below example tells that if we remove the rule with match with trailing slash, it gives different results (rewrites "/prefix/ok" to "//ok").